optuna/pruners/nop.py

```diff
@@ -11,16 +11,42 @@ class NopPruner(BasePruner):
 
     Example:
 
-        .. code::
-
-            >>> from optuna import create_study
-            >>> from optuna.pruners import NopPruner
-            >>>
-            >>> def objective(trial):
-            >>>     ...
-            >>>
-            >>> study = create_study(pruner=NopPruner())
-            >>> study.optimize(objective)
+        .. testsetup::
+
+            import numpy as np
+            from sklearn.model_selection import train_test_split
+
+            np.random.seed(seed=0)
+            X = np.random.randn(200).reshape(-1, 1)
+            y = np.where(X[:, 0] < 0.5, 0, 1)
+            X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
+            classes = np.unique(y)
+
+        .. testcode::
+
+            import optuna
+            from sklearn.linear_model import SGDClassifier
+
+            def objective(trial):
+                alpha = trial.suggest_uniform('alpha', 0.0, 1.0)
+                clf = SGDClassifier(alpha=alpha)
+                n_train_iter = 100
+
+                for step in range(n_train_iter):
+                    clf.partial_fit(X_train, y_train, classes=classes)
+
+                    intermediate_value = clf.score(X_test, y_test)
+                    trial.report(intermediate_value, step)
+
+                    if trial.should_prune():
+                        print("Never reach here")
```

Comment:
How about a slightly stricter doctest?
```suggestion
                    assert False, "should_prune() should always return False with this pruner."
```